lib/Dialect/SV/SVOps.cpp

```diff
@@ -137,16 +137,26 @@ void IfDefOp::build(OpBuilder &odsBuilder, OperationState &result,
 // IfOp
 
 void IfOp::build(OpBuilder &odsBuilder, OperationState &result, Value cond,
-                 std::function<void()> bodyCtor) {
+                 std::function<void()> thenCtor,
+                 std::function<void()> elseCtor) {
   result.addOperands(cond);
   Region *body = result.addRegion();
   IfOp::ensureTerminator(*body, odsBuilder, result.location);
 
   // Fill in the body of the #ifdef.
-  if (bodyCtor) {
+  if (thenCtor) {
     auto oldIP = &*odsBuilder.getInsertionPoint();
     odsBuilder.setInsertionPointToStart(&*body->begin());
-    bodyCtor();
+    thenCtor();
+    odsBuilder.setInsertionPoint(oldIP);
+  }
+
+  Region *elseRegion = result.addRegion();
```

Comment:
Seems like this builder always creates an empty else region, and the `hasElse` logic checks for emptiness. Maybe if there is no else constructor, just don't create a region in the builder here, and `hasElse` can check for null instead of an empty region? Not a big deal, but it seems a little weird to me to add an empty region rather than just not adding a region.